src/everest/config/everest_config.py

```diff
@@ -588,36 +592,36 @@ def simulation_dir(self) -> str | None:
         assert self.environment is not None
         path = self.environment.simulation_folder
 
-        if os.path.isabs(path):
+        if os.path.isabs(path):  # type: ignore
             return path
 
         cfgdir = self.output_dir
         if cfgdir is None:
             return path
 
-        return os.path.join(cfgdir, path)
+        return os.path.join(cfgdir, path)  # type: ignore
 
-    def _get_output_subdirectory(self, subdirname: str):
+    def _get_output_subdirectory(self, subdirname: str) -> str | None:
```

Comment:
This cannot be None because `self.output_dir` is never None. I think you can remove 
``` python 
  if self.output_dir is None:
      return None
```